rmf_demos_tasks/rmf_demos_tasks/dispatch_delivery.py

```diff
@@ -16,30 +16,41 @@
 
 import sys
 import uuid
-import time
 import argparse
+import json
+import asyncio
 
 import rclpy
 from rclpy.node import Node
 from rclpy.parameter import Parameter
-from rmf_task_msgs.srv import SubmitTask
-from rmf_task_msgs.msg import TaskType, Delivery
+from rclpy.qos import qos_profile_system_default
+from rclpy.qos import QoSProfile
+from rclpy.qos import QoSHistoryPolicy as History
+from rclpy.qos import QoSDurabilityPolicy as Durability
+from rclpy.qos import QoSReliabilityPolicy as Reliability
+
+from rmf_task_msgs.msg import ApiRequest, ApiResponse
 
-###############################################################################
 
+###############################################################################
 
-class TaskRequester:
+class TaskRequester(Node):
 
     def __init__(self, argv=sys.argv):
+        super().__init__('task_requester')
         parser = argparse.ArgumentParser()
-        parser.add_argument('-p', '--pickup', required=True,
-                            type=str, help='Start waypoint')
-        parser.add_argument('-pd', '--pickup_dispenser', required=True,
-                            type=str, help='Pickup dispenser name')
-        parser.add_argument('-d', '--dropoff', required=True,
-                            type=str, help='Finish waypoint')
-        parser.add_argument('-di', '--dropoff_ingestor', required=True,
-                            type=str, help='Dropoff ingestor name')
+        parser.add_argument('-p', '--pickups', required=True,
```

Comment:
It's great that we're trying to support multiple pickup/drop-offs. But I find packing the pickup and dropoff params into a string which is then split at `,` to be very un-intuitive. Suggest to leave the original args ( -p, -pd, -d, -di) and add any others that you may require (sku, qty etc). Then have these args accept arrays via `nargs='+'`. Within the script you check if the lengths of all the arrays are equal and if so add pickup and dropoff descriptions with values at corresponding indices. If there is a mismatch in the array lengths, we can print an error message for the user.

Reply:
I have changed the format of the script. An example is shown in the readme. 
  ```bash
  ros2 run rmf_demos_tasks dispatch_delivery \
  -p pantry pantry \
  -ph coke_dispenser coke_dispenser_2 \
  -d hardware_2 coe \
  -dh coke_ingestor coke_ingestor_2 \
  -pp coke,1 coke,1 \
  -dp coke,1 coke,1 \
  --use_sim_time
  ```

Within the script, it will conduct assertion checks.